TerrorConsole/Assets/_Main/Scripts/LevelManager/LevelManager.cs

```diff
@@ -49,9 +41,17 @@ public LevelState GetCurrentState()
             return currentLevelState;
         }
 
-        public SaveLevelData GetSaveLevelData()
+        private SaveLevelData GetSaveLevelData()
+        {
+            return SaveSystemManager.Source.GetLoadedGame().GetLevelData(levelNumber);
+        }
+        public void AddOrUpdateLevelEvent(string eventName, bool eventState)
+        {
+            GetSaveLevelData().AddOrUpdateLevelEvent(eventName, eventState);
```

Comment:
This is still wrong, you are still breaking encapsultation, we can fix this tonight if you want, `LevelManager` should never have access to the data structure the save system uses, it can only modify it if the `SaveSystem` has the methods for it, you need to add them.